src/providers/ad/ad_pac_common.c

```diff
@@ -215,10 +215,26 @@ errno_t check_upn_and_sid_from_user_and_pac(struct ldb_message *msg,
             DEBUG(SSSDBG_MINOR_FAILURE, "User object does not have a UPN but PAC "
                       "says otherwise, maybe ldap_user_principal option is set.\n");
             if (pac_check_opts & CHECK_PAC_CHECK_UPN) {
-                DEBUG(SSSDBG_CRIT_FAILURE,
-                      "UPN is missing but PAC UPN check required, "
-                      "PAC validation failed.\n");
-                return ERR_CHECK_PAC_FAILED;
+                if (pac_check_opts & CHECK_PAC_CHECK_UPN_ALLOW_MISSING) {
+                    DEBUG(SSSDBG_CRIT_FAILURE,
+                          "UPN is missing but PAC UPN check required, "
```

Comment:
If the option is set to allow missing UPN, should not be the message less critical - like IMPORTANT_INFO? This will trigger the back-trace in the log and I'm not sure if we want it.

What do you think?
T.

Reply:
Imo, IMPORTANT_INFO in the log + sss_log(SSS_LOG_CRIT) would be enough.

Reply:
Hi,

the idea behind `SSSDBG_CRIT_FAILURE` was to make the message annoying so that some action will be taken to make the message go away. But I'm fine with `SSSDBG_IMPORTANT_INFO`.

bye,
Sumit

Reply:
I see. Makes sense.